akka-remote-tests/src/multi-jvm/scala/akka/remote/RemoteRestartedQuarantinedSpec.scala

```diff
@@ -0,0 +1,139 @@
+    def receive = {
+      case "shutdown" ⇒ context.system.shutdown()
+      case "identify" ⇒ sender() ! (AddressUidExtension(context.system).addressUid, self)
+    }
+  }
+
+}
+
+class RemoteRestartedQuarantinedSpecMultiJvmNode1 extends RemoteRestartedQuarantinedSpec
+class RemoteRestartedQuarantinedSpecMultiJvmNode2 extends RemoteRestartedQuarantinedSpec
+
+abstract class RemoteRestartedQuarantinedSpec
+  extends MultiNodeSpec(RemoteRestartedQuarantinedSpec)
+  with STMultiNodeSpec with ImplicitSender {
+
+  import RemoteRestartedQuarantinedSpec._
+
+  override def initialParticipants = 2
+
+  def identifyWithUid(role: RoleName, actorName: String): (Int, ActorRef) = {
+    system.actorSelection(node(role) / "user" / actorName) ! "identify"
+    expectMsgType[(Int, ActorRef)]
+  }
+
+  "A restarted quarantined system" must {
+
+    "should not crash the other system (#17213)" taggedAs LongRunningTest in {
+
+      system.actorOf(Props[Subject], "subject")
+      enterBarrier("subject-started")
+
+      runOn(first) {
+        val secondAddress = node(second).address
+
+        val (uid, ref) = identifyWithUid(second, "subject")
+
+        RARP(system).provider.transport.quarantine(node(second).address, Some(uid))
+
+        enterBarrier("quarantined")
+        enterBarrier("still-quarantined")
+
+        testConductor.shutdown(second).await
+
+        within(30.seconds) {
+          awaitAssert {
+            system.actorSelection(RootActorPath(secondAddress) / "user" / "subject") ! Identify("subject")
+            expectMsgType[ActorIdentity](1.second).ref.get
+          }
+        }
+
+        system.actorSelection(RootActorPath(secondAddress) / "user" / "subject") ! "shutdown"
+      }
+
+      runOn(second) {
+        val addr = system.asInstanceOf[ExtendedActorSystem].provider.getDefaultAddress
+        val firstAddress = node(first).address
+
+        val (_, ref) = identifyWithUid(first, "subject")
+
+        enterBarrier("quarantined")
+
+        // Check that quarantine is intact
+        within(10.seconds) {
+          awaitAssert {
+            EventFilter.warning(pattern = "The remote system has quarantined this system", occurrences = 1).intercept {
+              ref ! "boo!"
+            }
+          }
+        }
+
+        enterBarrier("still-quarantined")
+
+        system.awaitTermination(10.seconds)
+
+        val freshSystem = ActorSystem(system.name, ConfigFactory.parseString(s"""
+                    akka.remote.retry-gate-closed-for = 0.5 s
+                    akka.remote.netty.tcp {
+                      hostname = ${addr.host.get}
+                      port = ${addr.port.get}
+                    }
+                    """).withFallback(system.settings.config))
+
+        val probe = TestProbe()(freshSystem)
+
+        freshSystem.actorSelection(RootActorPath(firstAddress) / "user" / "subject").tell(Identify("subject"), probe.ref)
+        probe.expectMsgType[ActorIdentity](1.second).ref.get
```

Comment:
I have tried that, but unfortunately it does not help.
Also, why would it not be possible to send a message from a fresh system? What is blocking that message?

This is the log when I repeatedly send:

    [second] # sending to subject
    [second] [DEBUG] [04/22/2015 19:13:30.701] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-6] [akka.serialization.Serialization(akka://RemoteRestartedQuarantinedSpec)] Using serializer[akka.serialization.JavaSerializer] for message [akka.remote.RemoteWatcher$ReapUnreachableTick$]
    [second] [DEBUG] [04/22/2015 19:13:30.701] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-5] [akka.serialization.Serialization(akka://RemoteRestartedQuarantinedSpec)] Using serializer[akka.serialization.JavaSerializer] for message [akka.remote.RemoteWatcher$HeartbeatTick$]
    [second] [DEBUG] [04/22/2015 19:13:30.706] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-6] [akka.serialization.Serialization(akka://RemoteRestartedQuarantinedSpec)] Using serializer[akka.serialization.JavaSerializer] for message [akka.remote.transport.ThrottledAssociation$Dequeue$]
    [second] [DEBUG] [04/22/2015 19:13:30.709] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-5] [akka.serialization.Serialization(akka://RemoteRestartedQuarantinedSpec)] Using serializer[akka.serialization.JavaSerializer] for message [akka.remote.ReliableDeliverySupervisor$GotUid]
    [second] [DEBUG] [04/22/2015 19:13:30.710] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-5] [akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@second/system/endpointManager/reliableEndpointWriter-akka.trttl.gremlin.tcp%3A%2F%2FRemoteRestartedQuarantinedSpec%40localhost%3A50560-0/endpointWriter] sending message RemoteMessage: [ActorSelectionMessage(Identify(subject),Vector(user, subject),false)] to [Actor[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/]]<+[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/] from [Actor[akka://RemoteRestartedQuarantinedSpec/system/testActor2#815004902]]
    [second] [DEBUG] [04/22/2015 19:13:30.710] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-5] [akka.serialization.Serialization(akka://RemoteRestartedQuarantinedSpec)] Using serializer[akka.serialization.JavaSerializer] for message [akka.actor.Identify]
    [second] [DEBUG] [04/22/2015 19:13:30.711] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-5] [akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@second/system/endpointManager/reliableEndpointWriter-akka.trttl.gremlin.tcp%3A%2F%2FRemoteRestartedQuarantinedSpec%40localhost%3A50560-0/endpointWriter] Drained buffer with maxWriteCount: 50, fullBackoffCount: 1, smallBackoffCount: 0, noBackoffCount: 0 , adaptiveBackoff: 1000
    [second] [DEBUG] [04/22/2015 19:13:30.871] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-5] [akka.serialization.Serialization(akka://RemoteRestartedQuarantinedSpec)] Using serializer[akka.serialization.JavaSerializer] for message [akka.remote.EndpointWriter$AckIdleCheckTimer$]
    [first] [DEBUG] [04/22/2015 19:13:31.640] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-5] [akka.serialization.Serialization(akka://RemoteRestartedQuarantinedSpec)] Using serializer[akka.serialization.JavaSerializer] for message [akka.remote.ReliableDeliverySupervisor$AttemptSysMsgRedelivery$]
    [second] # sending to subject
    [second] [DEBUG] [04/22/2015 19:13:32.522] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-5] [akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@second/system/endpointManager/reliableEndpointWriter-akka.trttl.gremlin.tcp%3A%2F%2FRemoteRestartedQuarantinedSpec%40localhost%3A50560-0/endpointWriter] sending message RemoteMessage: [ActorSelectionMessage(Identify(subject),Vector(user, subject),false)] to [Actor[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/]]<+[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/] from [Actor[akka://RemoteRestartedQuarantinedSpec/system/testActor3#-510013953]]
    [second] [DEBUG] [04/22/2015 19:13:32.722] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-6] [akka.serialization.Serialization(akka://RemoteRestartedQuarantinedSpec)] Using serializer[akka.serialization.JavaSerializer] for message [akka.remote.ReliableDeliverySupervisor$AttemptSysMsgRedelivery$]
    [second] # sending to subject
    [second] [DEBUG] [04/22/2015 19:13:34.342] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-6] [akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@second/system/endpointManager/reliableEndpointWriter-akka.trttl.gremlin.tcp%3A%2F%2FRemoteRestartedQuarantinedSpec%40localhost%3A50560-0/endpointWriter] sending message RemoteMessage: [ActorSelectionMessage(Identify(subject),Vector(user, subject),false)] to [Actor[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/]]<+[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/] from [Actor[akka://RemoteRestartedQuarantinedSpec/system/testActor4#-1976802455]]
    [second] # sending to subject
    [second] [DEBUG] [04/22/2015 19:13:36.162] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-6] [akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@second/system/endpointManager/reliableEndpointWriter-akka.trttl.gremlin.tcp%3A%2F%2FRemoteRestartedQuarantinedSpec%40localhost%3A50560-0/endpointWriter] sending message RemoteMessage: [ActorSelectionMessage(Identify(subject),Vector(user, subject),false)] to [Actor[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/]]<+[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/] from [Actor[akka://RemoteRestartedQuarantinedSpec/system/testActor5#-173904844]]
    [second] # sending to subject
    [second] [DEBUG] [04/22/2015 19:13:37.979] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-6] [akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@second/system/endpointManager/reliableEndpointWriter-akka.trttl.gremlin.tcp%3A%2F%2FRemoteRestartedQuarantinedSpec%40localhost%3A50560-0/endpointWriter] sending message RemoteMessage: [ActorSelectionMessage(Identify(subject),Vector(user, subject),false)] to [Actor[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/]]<+[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/] from [Actor[akka://RemoteRestartedQuarantinedSpec/system/testActor6#-820510536]]
    [second] # sending to subject
    [second] [DEBUG] [04/22/2015 19:13:39.799] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-6] [akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@second/system/endpointManager/reliableEndpointWriter-akka.trttl.gremlin.tcp%3A%2F%2FRemoteRestartedQuarantinedSpec%40localhost%3A50560-0/endpointWriter] sending message RemoteMessage: [ActorSelectionMessage(Identify(subject),Vector(user, subject),false)] to [Actor[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/]]<+[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/] from [Actor[akka://RemoteRestartedQuarantinedSpec/system/testActor7#2043750267]]
    [second] # sending to subject
    [second] [DEBUG] [04/22/2015 19:13:41.619] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-5] [akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@second/system/endpointManager/reliableEndpointWriter-akka.trttl.gremlin.tcp%3A%2F%2FRemoteRestartedQuarantinedSpec%40localhost%3A50560-0/endpointWriter] sending message RemoteMessage: [ActorSelectionMessage(Identify(subject),Vector(user, subject),false)] to [Actor[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/]]<+[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/] from [Actor[akka://RemoteRestartedQuarantinedSpec/system/testActor8#1748866962]]
    [second] # sending to subject
    [second] [DEBUG] [04/22/2015 19:13:43.439] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-5] [akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@second/system/endpointManager/reliableEndpointWriter-akka.trttl.gremlin.tcp%3A%2F%2FRemoteRestartedQuarantinedSpec%40localhost%3A50560-0/endpointWriter] sending message RemoteMessage: [ActorSelectionMessage(Identify(subject),Vector(user, subject),false)] to [Actor[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/]]<+[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/] from [Actor[akka://RemoteRestartedQuarantinedSpec/system/testActor9#-1857711870]]
    [second] # sending to subject
    [second] [DEBUG] [04/22/2015 19:13:45.258] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-5] [akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@second/system/endpointManager/reliableEndpointWriter-akka.trttl.gremlin.tcp%3A%2F%2FRemoteRestartedQuarantinedSpec%40localhost%3A50560-0/endpointWriter] sending message RemoteMessage: [ActorSelectionMessage(Identify(subject),Vector(user, subject),false)] to [Actor[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/]]<+[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/] from [Actor[akka://RemoteRestartedQuarantinedSpec/system/testActor10#1081143764]]
    [second] # sending to subject
    [second] [DEBUG] [04/22/2015 19:13:47.078] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-5] [akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@second/system/endpointManager/reliableEndpointWriter-akka.trttl.gremlin.tcp%3A%2F%2FRemoteRestartedQuarantinedSpec%40localhost%3A50560-0/endpointWriter] sending message RemoteMessage: [ActorSelectionMessage(Identify(subject),Vector(user, subject),false)] to [Actor[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/]]<+[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/] from [Actor[akka://RemoteRestartedQuarantinedSpec/system/testActor11#1318566221]]
    [second] # sending to subject
    [second] [DEBUG] [04/22/2015 19:13:48.894] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-6] [akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@second/system/endpointManager/reliableEndpointWriter-akka.trttl.gremlin.tcp%3A%2F%2FRemoteRestartedQuarantinedSpec%40localhost%3A50560-0/endpointWriter] sending message RemoteMessage: [ActorSelectionMessage(Identify(subject),Vector(user, subject),false)] to [Actor[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/]]<+[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/] from [Actor[akka://RemoteRestartedQuarantinedSpec/system/testActor12#-1025352373]]
    [first] [INFO] [04/22/2015 19:13:49.643] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-5] [akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/system/transports/akkaprotocolmanager.trttl.gremlin.tcp0/akkaProtocol-trttl.gremlin.tcp%3A%2F%2FRemoteRestartedQuarantinedSpec%40localhost%3A50561-4] No response from remote. Handshake timed out or transport failure detector triggered.
    [first] [DEBUG] [04/22/2015 19:13:49.644] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-6] [akka.serialization.Serialization(akka://RemoteRestartedQuarantinedSpec)] Using serializer[akka.serialization.JavaSerializer] for message [akka.actor.Status$Failure]
    [first] [WARN] [04/22/2015 19:13:49.644] [RemoteRestartedQuarantinedSpec-akka.actor.default-dispatcher-14] [akka://RemoteRestartedQuarantinedSpec/system/transports/throttlermanager.gremlin.tcp0/throttler4] received dead letter from Actor[akka://RemoteRestartedQuarantinedSpec/deadLetters]: PoisonPill
    [first] [INFO] [04/22/2015 19:13:49.644] [RemoteRestartedQuarantinedSpec-akka.actor.default-dispatcher-3] [akka://RemoteRestartedQuarantinedSpec/system/transports/throttlermanager.gremlin.tcp0/throttler4] Message [akka.actor.PoisonPill$] from Actor[akka://RemoteRestartedQuarantinedSpec/deadLetters] to Actor[akka://RemoteRestartedQuarantinedSpec/system/transports/throttlermanager.gremlin.tcp0/throttler4#270646588] was not delivered. [7] dead letters encountered. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
    [first] [WARN] [04/22/2015 19:13:49.653] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-5] [akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/system/endpointManager/reliableEndpointWriter-akka.trttl.gremlin.tcp%3A%2F%2FRemoteRestartedQuarantinedSpec%40localhost%3A50561-2] Association with remote system [akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@second] has failed, address is now gated for [1000] ms. Reason: [Association failed with [akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@second]] Caused by: [No response from remote. Handshake timed out or transport failure detector triggered.]
    [first] [DEBUG] [04/22/2015 19:13:49.657] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-5] [akka.serialization.Serialization(akka://RemoteRestartedQuarantinedSpec)] Using serializer[akka.serialization.JavaSerializer] for message [akka.actor.ActorIdentity]
    [second] # sending to subject
    [second] [DEBUG] [04/22/2015 19:13:50.714] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-6] [akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@second/system/endpointManager/reliableEndpointWriter-akka.trttl.gremlin.tcp%3A%2F%2FRemoteRestartedQuarantinedSpec%40localhost%3A50560-0/endpointWriter] sending message RemoteMessage: [ActorSelectionMessage(Identify(subject),Vector(user, subject),false)] to [Actor[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/]]<+[akka.trttl.gremlin.tcp://RemoteRestartedQuarantinedSpec@first/] from [Actor[akka://RemoteRestartedQuarantinedSpec/system/testActor13#-2011911342]]
    [second] [DEBUG] [04/22/2015 19:13:51.271] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-5] [akka.serialization.Serialization(akka://RemoteRestartedQuarantinedSpec)] Using serializer[akka.serialization.JavaSerializer] for message [akka.remote.EndpointWriter$StopReading]
    [second] [DEBUG] [04/22/2015 19:13:51.272] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-5] [akka.serialization.Serialization(akka://RemoteRestartedQuarantinedSpec)] Using serializer[akka.serialization.JavaSerializer] for message [akka.remote.ReliableDeliverySupervisor$Ungate$]
    [second] [DEBUG] [04/22/2015 19:13:51.272] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-6] [akka.serialization.Serialization(akka://RemoteRestartedQuarantinedSpec)] Using serializer[akka.serialization.JavaSerializer] for message [akka.actor.UnhandledMessage]
    [second] [DEBUG] [04/22/2015 19:13:51.273] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-5] [akka.serialization.Serialization(akka://RemoteRestartedQuarantinedSpec)] Using serializer[akka.serialization.JavaSerializer] for message [akka.remote.EndpointWriter$StoppedReading]
    [first] [DEBUG] [04/22/2015 19:13:51.273] [RemoteRestartedQuarantinedSpec-akka.remote.default-remote-dispatcher-5] [akka.trttl.gremlin.